contracts/src/Consensus.sol

```diff
@@ -0,0 +1,460 @@
+    - Ownable (OpenZeppelin): For owner-controlled threshold updates
+
+    ============================================================
+    STRUCTS
+    ============================================================
+
+    1. SafeTransaction
+       - to: address                 // Target address for the Safe transaction
+       - value: uint256              // ETH value to send
+       - data: bytes                 // Transaction data payload
+       - operation: uint8            // 0 = Call, 1 = DelegateCall
+       - safeTxGas: uint256          // Gas for the Safe transaction
+       - baseGas: uint256            // Base gas (e.g., for signatures)
+       - gasPrice: uint256           // Gas price for refund calculation
+       - gasToken: address           // Token for gas refund (address(0) for ETH)
+       - refundReceiver: address     // Address receiving gas refund
+       - nonce: uint256              // Safe transaction nonce
+       - chainId: uint256            // Chain ID where transaction will be executed
+       - epoch: uint256              // Epoch number when transaction was proposed
+       - proposer: address           // Address that proposed this transaction
+       - safeAddress: address        // Safe wallet address this transaction is for
+       - timestamp: uint256          // Block timestamp when transaction was proposed
+
+    2. ConfigProposal
+       - value: uint128              // Proposed threshold value
+       - executableAt: uint128       // Timestamp when proposal can be executed (0 if none)
+
+    ============================================================
+    STORAGE VARIABLES
+    ============================================================
+
+    State Variables:
+    - stakingContract: address (immutable)
+        // Reference to Staking contract for validator verification
+
+    - frostCoordinator: address (immutable)
+        // Reference to FROSTCoordinator for KeyGen ceremonies
+
+    - currentEpoch: uint256
+        // Current epoch number (incremented with each KeyGen)
+        // Starts at 1 (first epoch initiated in constructor)
+
+    - lastEpochTimestamp: uint256
+        // Timestamp of the last epoch increment
+        // Used to enforce 86400 second (1 day) minimum between epochs
+
+    - threshold: uint128
+        // Minimum number of validators required for FROST signing
+        // Must satisfy: 1 < threshold <= validator count
+        // Updated through time-locked proposal mechanism
+
+    - pendingThresholdChange: ConfigProposal
+        // Pending threshold change proposal with timelock
+
+    - CONFIG_TIME_DELAY: uint256 (immutable)
+        // Time delay for threshold changes (e.g., 7 days)
+        // Set at deployment, same pattern as Staking contract
+
+    Mappings:
+    - registeredValidators: mapping(address => bool)
+        // Tracks validators registered for the NEXT epoch
+        // Validators can register at any time, included in next KeyGen
```

Comment:
I'm concerned that we already have a similar mapping in the staking contract. I don't like keeping track of the same state in two different places.